### PR TITLE
Create flow for building a fresh solr instance from a dump file

### PR DESCRIPTION
**Pull request by cdrini**
Progress towards #1067 (and, in general, all our solr issues).

**Description**:
Sorry folks, this a very messy PR, but ran out of time to make this neater. I think there is a _lot_ of room for improvement with this PR (more tests, general cleanup, etc), but I think it is ready to merge. It's been run multiple times with few errors.

**Technical**:
General "How it works": A Jenkins job loads the dump file into a local postgres instance, import works/orphans, and then imports them into solr. Solr needs to query the data as its importing; that's why the db is needed.

**Testing**:
Have run it multiple times on OJF; it does have issues on occasion, but most issues are recoverable. Since this is more of a dev ops tool anyways, I'm ok with that. http://server.openjournal.foundation:8081/job/solrbuilder-reindex/

### CODE REVIEW

**Comment by mekarpeles** [comment]
Is this still WIP?

**Comment by cdrini** [comment]
Yes, unfortunately; have some final cleanup left.

**Comment by mekarpeles** [comment]
@cdrini I see this issue as WIP -- should you be assigned? Should this be marked as WIP?

**Comment by tfmorris** [comment]
So is this getting close to being ready for review? I see that the WIP tag got removed.

I have a couple of dozen comments from a review that I began 14 months ago, but I imagine many of them are obsolete. I don't want to go to the trouble of re-reviewing them and updating them though, if this is still in flux.

When we last chatted 9 months ago there was basic disagreement on whether having this be simple, fast, and runnable by any developer were important goals. Has any of that changed? I believe these attributes are critical to being able to iterate quickly by multiple developers. The PR #2246 attempts to improve along these dimensions, but it hasn't been looked at it in a while.

**Comment by tfmorris** [comment]
@cdrini Ping. Any feedback on my question from a week and a half ago?

File: scripts/solr_builder/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+      dir 'scripts/solr_builder'
+      // Needed to make docker-in-docker work
+      args '-v /var/run/docker.sock:/var/run/docker.sock'
+      args '-v /var/lib/docker/volumes/jenkins-data:/var/lib/docker/volumes/jenkins-data'
+      // Where to store dumps
+      args '-v /storage:/storage'
+    }
+  }
+  parameters {
+    string(name: 'SOLR_BACKUP_GZIP', defaultValue: 'backup-2019-07-30.tar.gz', description: 'Location of the dump gzip in /storage/openlibrary/solr')
```
Comment (mekarpeles) [inline scripts/solr_builder/Jenkinsfile:14]: is it okay to hard-code against `/storage` which is OJF specific?

File: scripts/solr_builder/solr_builder/solr_builder_main.py [hunk 2]
```diff
@@ -0,0 +1,474 @@
+from __future__ import division
+
+import ConfigParser
```
Comment (mekarpeles) [inline scripts/solr_builder/solr_builder/solr_builder_main.py:3]: check python2 v. python3 here

File: scripts/solr_builder/solr_builder/solr_builder_main.py [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+def config_section_to_dict(config_file, section):
+    """
+    Read a config file's section as a dict
+
+    :param str config_file: filename of config file
+    :param str section: section to pull data from
+    :return: dict of key value pairs
+    :rtype: dict
+    """
+    config = ConfigParser.ConfigParser()
```
Comment (mekarpeles) [inline scripts/solr_builder/solr_builder/solr_builder_main.py:29]: same re: py2v.3

**Pull request merged**